aten/src/ATen/detail/MPSHooksInterface.cpp

```diff
@@ -0,0 +1,35 @@
+//  Copyright © 2022 Apple Inc.
+
+#include <ATen/detail/MPSHooksInterface.h>
+#include <c10/util/Exception.h>
+#include <c10/util/CallOnce.h>
+
+namespace at {
+namespace detail {
+
+const MPSHooksInterface& getMPSHooks() {
+  static std::unique_ptr<MPSHooksInterface> mps_hooks;
+#if !defined C10_MOBILE
+  static c10::once_flag once;
+  c10::call_once(once, [] {
+    mps_hooks = MPSHooksRegistry()->Create("MPSHooks", MPSHooksArgs{});
+    if (!mps_hooks) {
+      mps_hooks =
+          // NOLINTNEXTLINE(modernize-make-unique)
+          std::unique_ptr<MPSHooksInterface>(new MPSHooksInterface());
+    }
+  });
+#else
+  if (mps_hooks == nullptr) {
+    mps_hooks =
+        // NOLINTNEXTLINE(modernize-make-unique)
+        std::unique_ptr<MPSHooksInterface>(new MPSHooksInterface());
```

Comment:
Indeed, modernize your C++ :)
```suggestion
    mps_hooks = std::make_unique<MPSHooksInterface>();
```